src/MeshlessFV/MeshlessFVTree.cpp

```diff
@@ -491,6 +491,8 @@ void MeshlessFVTree<ndim,ParticleType,TreeCell>::UpdateGradientMatrices
         // Make local copy of hmask for active particle
         hmask = mfv->types[activepart[j].ptype].hmask;
 
+        activepart[j].levelneib = 0;
+
```

Comment:
It's what is done in the SPH code too, but I think it is correct. The code does the following:
1) Calculates levelneib for active particles
2) Increases levelneib for inactive particles, if any of the newly active particles have a higher level (smaller time step) than the inactive particles had before.

We don't need to worry about the case that levelneib actually decreases for inactive particles, although it might allow us to increase their timestep if we did. But I'm not aware of anyone doing that since the main purpose is to detect and reduce the timestep when in the region of strong shocks.